Dockerfile

```diff
@@ -97,7 +97,7 @@ ENV PATH $PYTHONUSERBASE/bin:$PATH
 ENV NPM_CONFIG_PREFIX=/deps/
 RUN ln -s ${HOME}/package.json /deps/package.json \
     && ln -s ${HOME}/package-lock.json /deps/package-lock.json \
-    && make update_deps
+    && make update_deps --prod
```

Comment:
should be 

```bash
make update_deps ARGS="--prod"
```

Reply:
I also noticed that because we are copying the root directory before running update_deps, we might not be allowing docker the ability to cache deps across builds. I would just fix this now, unless you think it really should go to a different PR?

Reply:
> I also noticed that because we are copying the root directory before running update_deps, we might not be allowing docker the ability to cache deps across builds. I would just fix this now, unless you think it really should go to a different PR?

Could you expand on this? It's not immediately clear what you're proposing.

Reply:
The way we organize our dockerfile is not the most efficient when it comes to taking advantage of how docker caches layers. We copy our source code too early. 

I tried getting this to work, it can be done but it is definitely too much for this PR. I'll open a separate ticket for this explaining more.

Reply:
Also just noting, I wonder if adding those "shell" dependencies to prod.txt is the best solution long term. I think having a dedicated instance of addons-server which has prod/dev installed might be better since we don't need to bloat all the prod images with stuff that only needs to be available in one place.

WE could even not build a separate image but in the deployment install the dev depdencies. Then if you want to shell, you go to the "shell" pod and not web or worker. Just a thought but I think it would save us space on the images and time on the builds.

Reply:
Probably better from a ergonomics / security perspective as we never ssh into web or worker pods, but only the dedicated shell pods.

Reply:
Yes, typically in prod SRE will only spawn django shells from a dedicated admin pod I think - we're less strict in dev/stage. Down the line it's a good idea (that's why `dev.txt` was separate in the first place) but out of scope for now.

Reply:
I get this:

```bash
ipython==8.20.0
├── decorator [required: Any, installed: 5.1.1]
├── exceptiongroup [required: Any, installed: 1.2.0]
├── jedi [required: >=0.16, installed: 0.19.1]
│   └── parso [required: >=0.8.3,<0.9.0, installed: 0.8.3]
├── matplotlib-inline [required: Any, installed: ?]
├── pexpect [required: >4.3, installed: 4.9.0]
│   └── ptyprocess [required: >=0.5, installed: 0.7.0]
├── prompt-toolkit [required: >=3.0.41,<3.1.0, installed: 3.0.43]
│   └── wcwidth [required: Any, installed: 0.2.13]
├── pygments [required: >=2.4.0, installed: 2.17.2]
├── stack-data [required: Any, installed: 0.6.3]
│   ├── asttokens [required: >=2.1.0, installed: 2.4.1]
│   │   └── six [required: >=1.12.0, installed: 1.16.0]
│   ├── executing [required: >=1.2.0, installed: 2.0.1]
│   └── pure-eval [required: Any, installed: 0.2.2]
└── traitlets [required: >=5, installed: 5.14.1]
```

I moved everything that already existed into prod.txt, several deps were not even specified in either .txt (Iguess that's fine?>..) 

managing python dependencies is like playing piñata without the candy.

Reply:
We've purposefully ignored `matplotlib-inline` because we shouldn't need it for our usage. Everything else is already specified in requirements.